app/controllers/api/promos_controller.rb

```diff
@@ -4,21 +4,12 @@ class Api::PromosController < Api::BaseController
 
   api_versions :v1
 
-  def resource
-    super.tap do |af|
-      if story && af.try(:new_record?)
-        af.audio_version_id ||= story.promos.id
-        af.account_id ||= story.account_id
```

Comment:
Basically, it's because there is no other way to get to the promos
controller than through a story, and the resources_base is what will be
built on. So instead of AudioFile.new, the controller is calling
story.promo_audio.build
On Apr 28, 2015 11:01, "Andrew Kuklewicz" <notifications@github.com> wrote:

> In app/controllers/api/promos_controller.rb
> <https://github.com/PRX/cms.prx.org/pull/103#discussion_r29251389>:
>
> > @@ -4,21 +4,12 @@ class Api::PromosController < Api::BaseController
> >
> >    api_versions :v1
> >
> > -  def resource
> > -    super.tap do |af|
> > -      if story && af.try(:new_record?)
> > -        af.audio_version_id ||= story.promos.id
> > -        af.account_id ||= story.account_id
>
> So these are coming from the request params, provided by using a nested
> route to get to this call, right?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/PRX/cms.prx.org/pull/103/files#r29251389>.
>